module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,62 @@
+#include "AppendNewRecipe.hpp"
+
+#include <algorithm>
+#include <fstream>
+
+bool AppendNewRecipe(std::vector<std::string> steps,
+                     const std::list<std::string>& ingredients,
+                     const std::deque<std::pair<size_t, char>>& amount) {
+    std::ofstream file("recipes.txt", std::ios::app);
+    if (file.is_open()) {
+        auto text = FormatRecipit(steps, ingredients, amount).str();
+        file << text;
+        file.close();
+        return true;
+    }
+    return false;
+}
+
+std::vector<std::string> FormatIngredients(const std::list<std::string>& ingredients,
+                                           const std::deque<std::pair<size_t, char>>& amount) {
+    if (ingredients.size() != amount.size()) {
+        return {};
+    }
+    std::vector<std::string> formattedIngredients(ingredients.size());
+    std::transform(begin(ingredients),
+                   end(ingredients),
+                   begin(amount),
+                   begin(formattedIngredients),
+                   [](const auto& ingredient, const auto& pair) {
+                       auto [amount, unit] = pair;
+                       std::string formattedUnit{};
+                       if (unit == 'g') {
+                           formattedUnit = "gram";
+                       }
+                       if (unit == 's') {
+                           formattedUnit = "szklanka(i)";
+                       }
+                       if (unit == 'm') {
+                           formattedUnit = "mililitrow";
+                       }
+                       return std::to_string(amount) + ' ' + formattedUnit + ' ' + ingredient;
+                   });
+    return formattedIngredients;
+}
+
+std::stringstream FormatRecipit(std::vector<std::string> steps,
+                                const std::list<std::string>& ingredients,
+                                const std::deque<std::pair<size_t, char>>& amount) {
+    std::stringstream stream;
+    stream << "Skladniki:\n";
+    auto formattedIngredients = FormatIngredients(ingredients, amount);
+    for (auto line : formattedIngredients) {
+        stream << line << ",\n";
+    }
+    stream << '\n';
+    stream << "Kroki:\n";
+    for (size_t i = 0; i < steps.size(); ++i) {
+        stream << i + 1 << ") " << steps[i] << ".\n";
+    }
+    stream << "___________________________________\n";
```

Comment:
Create std::string(occurrences, char)

Reply:
I agree with you that it is much easier to create a dash like this, however:
In the tests you created this variable:
```cpp
constexpr const char kExpected[] =
    "Skladniki:\n20 gram cukru,\n1 szklanka(i) maki,"
    "\n40 mililitrow rumu,\n\nKroki:\n1) Wsypac do miski"
    " 20 gram cukru.\n2) Dorzucic 1 szklanke maki.\n3)"
    " Dokladnie wymieszac.\n4) Nalac 40ml rumu do kieliszka.\n"
    "5) Wypic kieliszek.\n6) Wysypac zawartosc miski."
    "\n___________________________________\n";
```
My output has to be equal to this text, so I just copied the dash, I was too lazy to count the number of dashes you put in kExpected.